pilot/pkg/networking/core/cluster_builder.go

```diff
@@ -529,7 +533,6 @@ func (cb *ClusterBuilder) buildDefaultPassthroughCluster() *cluster.Cluster {
 	}
 	cluster.AltStatName = util.DelimitedStatsPrefix(util.PassthroughCluster, cb.proxyVersion)
 	cb.applyConnectionPool(cb.req.Push.Mesh, newClusterWrapper(cluster), &networking.ConnectionPoolSettings{})
-	cb.applyMetadataExchange(cluster)
```

Comment:
I think this is not needed

Reply:
Do we support mesh workloads that aren't known by the client ahead of time? In other words, if I send traffic to a meshed VM, but my local control plane doesn't know about that VM, should it work? I think in most cases, the lack of ALPN would break it right?

Reply:
AFAIK ALPN and mTLS is used if we do know that the workload has a sidecar ( using the label ). So if we don't know - normally it will not be used (at least in permissive mode, with no DR changing defaults ).

For TCP - no ALPN means we don't send metadata. However it is not clear what happens if the external site does enable ALPN with istio special magic - I believe we would send the metatata ( that would happen if the external VM is malicious and wants to grab private data ). That would be a security/privacy issue and I don't think the http header setting helps. 

For HTTP - I believe we do send the metadata, unless the proxy config changes the headers.

Reply:
BTW - I am not sure about the behavior on MESH_EXTERNAL with TCP if the remote is (maliciously) doing ALPN. This PR would obviously make sure it's not leaking metadata - but I don't think we test this case ( or consider it ).